bindings/go/helm/input/blob.go

```diff
@@ -136,6 +145,69 @@ func newReadOnlyChart(path, tmpDirBase string) (result *ReadOnlyChart, err error
 	return result, nil
 }
 
+// newReadOnlyChartFromRemote downloads a chart from a remote Helm repository
+// and creates a ReadOnlyChart from it.
+func newReadOnlyChartFromRemote(ctx context.Context, helmSpec v1.Helm, tmpDirBase string, credentials map[string]string) (result *ReadOnlyChart, err error) {
+	settings := cli.New()
+	tmpDir, err := os.MkdirTemp(tmpDirBase, "helmRemoteChart*")
+	if err != nil {
+		return nil, fmt.Errorf("error creating temporary directory: %w", err)
+	}
+	defer func() {
+		if cerr := os.RemoveAll(tmpDir); cerr != nil {
+			err = errors.Join(err, fmt.Errorf("failed to cleanup temp dir: %w", cerr))
+		}
+	}()
+
+	dl := &downloader.ChartDownloader{
+		Out:              os.Stderr,
+		Verify:           downloader.VerifyNever, // TODO: Support signature verification
+		Getters:          getter.All(settings),
+		RepositoryConfig: settings.RepositoryConfig,
+		RepositoryCache:  settings.RepositoryCache,
+	}
+
+	// TODO: Hack.
+	if username, ok := credentials["username"]; ok {
+		if password, ok := credentials["password"]; ok {
+			dl.Options = append(dl.Options, getter.WithBasicAuth(username, password))
+		}
+	}
+
+	if helmSpec.CACert != "" || helmSpec.CACertFile != "" {
+		// TODO: Supporting CACert is difficult process since it could require a password.
```

Comment:
The SDK does actually not support out of the box certificate handling. I have to construct an HTTP client and pass that in. But that's only if I construct a full repository downloader. Ugh.